### DISCUSSION TITLE
FOS 7.50 7590 / LabBeta 7490 und MyFritz / USP-TR369 / Easy Support - Findings/Tunings

### DISCUSSION THREAD

**Discussion by maverick3005** [Freetz-NG/freetz-ng#741] [Allgemein]
Ich habe testweise mal die FOS 7.50 auf die 7590 gepackt und die aktuelle LabBeta auf eine 7490:

**Status 7490: Labor TESTED - working.**

Folgender kosmetischer Bug bei allen FOS 7.50/Beta, WENN im Freetz "MyFritz" "removed wurde:

![myfritz](https://user-images.githubusercontent.com/119104833/220303024-77dae089-37c1-4381-81eb-b1abca04345c.png)
Auf der Hauptübersichtsseite gibt es rechts noch den Menüpunkt:

"Täglich im Blick mit MyFritz"
das sollte durch einen Patch noch entfernt werden, wenn MyFritz "removed" wurde, da der Link sonst ins Leere rennt (cosmetic)

zweites Finding:

USP/TR369:

Bei der Migration von 7.29 -> 7.50 war auf der 7590 unter "Anbieterdienste/Internetzugang" noch ein Eintrag;
"USP/Anbieterdienste "anhaken". Dieses lies sich lediglich einmalig deaktivieren, seitdem erscheint dieser Menüpunkt nicht mehr.

Die Infos im "menuconfig" sollten dahingehend erweitert werden, das mit deaktivieren von TR069 ebenso TR369/USP mit deaktiviert wird. Nötig ist das für den Betrieb von MyFritz und einiger (neuer) Funktionen. (AVM Mobile Device Apps, EasySupport, etc.)

Aktuell scheint das wohl die Krux zu sein, das wenn man tr069 "removed" das dann auch USP mit deaktiviert wird. Hier könnte man ggfs. die Hilfe im "menuconfig" für die FOS 7.50 erweitern mit einem entspechenden Hinweis.

TR069 selber kann man über die ar7.cfg selbst bei laufendem tr069-Dienst notfalls noch auf disabled setzen. tr369/usp muss ich mir nochmal ansehen wie es da mit enable/disable dann in der Praxis aussieht über die neue tr369.cfg config.

Laut Beschreibung/Hilfe-Seiten soll es vom "Anbieter" abhängig sein, in wie weit USP dann unterstützt wird, im Falle von "Telekom" wäre das "EasySupport". Dabei ist mir aufgefallen das bei Nutzung eines externen Modems diese Funktion ja nicht mehr anwählbar ist.
Dürfte also unmittelbar mit dem entfernen von tr069 auch zusammenhängen ?

**Für mich sieht es jedenfalls so aus, das wenn "MyFritz" aktiv bleiben soll, muss auch tr069 weiterhin aktiv sein ?**

Evtl. könnte der Eine oder Andere hier auch noch mal einen Blick drauf werfen betreffs der genaueren Zusammenhänge zwischen diesen beiden Diensten ?

**Comment by maverick3005** [comment]
Lustiger Weise wird jedoch auf der Übersichtsseite "Diagnose" am Ende der Seite noch das hier angezeigt:
![diagnose](https://user-images.githubusercontent.com/119104833/220312534-99075c36-4155-4f87-a2cc-1038925cf6bb.png)

**Comment by fda77** [comment]
Hie ist die Patchdatei die die erweitern kannst: https://github.com/Freetz-NG/freetz-ng/blob/master/patches/scripts/221-remove_myfritz.sh zb mit eine `[ "$FREETZ_AVM_VERSION_07_5X_MIN" == "y" ]` block

> Bei der Migration von 7.29 -> 7.50 war auf der 7590 unter "Anbieterdienste/Internetzugang" noch ein Eintrag;
> "USP/Anbieterdienste "anhaken". Dieses lies sich lediglich einmalig deaktivieren, seitdem erscheint dieser Menüpunkt nicht mehr.

Sowas hatte ich auch, aber dann hatte ich es schon weggeclickt... vielleicht kannst du auch dazu einen patch machen dass es immer sichtbar ist? 

Wenn du auf der Fritzbox selbst Dateien editierst (zb mit dem "vix" Hilfsscript) kann es sein dass es im avm-gui nicht sofort sichtbar ist da es irgend eine art von cache-mechanismus gibt (restart von ctlmgr oder multid könntne helfen)

> das[s] wenn man tr069 "removed" das dann auch USP mit deaktiviert wird.

USP hab ich mir ehrlich gesagt nicht angeschaut, ich bin nur froh wenn die suspekten AVM dienste abgeschaltet sind

> Für mich sieht es jedenfalls so aus, das wenn "MyFritz" aktiv bleiben soll, muss auch tr069 weiterhin aktiv sein ?

Versteh ich nicht, das sind doch 2 verschiedene Dinge.

> wird jedoch auf der Übersichtsseite "Diagnose" am Ende der Seite noch das hier angezeigt:

An der Diagnose wird bis jetzt gar nichts gepatcht, ich denke das ist auch gut so. Bei mir ist da aber nichts:
![image](https://user-images.githubusercontent.com/471850/220336911-3f7e5416-82ab-4b5e-b2f2-fdf47b1e58e1.png)

**Comment by maverick3005** [comment]
Danke für den direkten Link erstmal. 

Ich hatte die Diskussion eigentlich allgemein mal aufgemacht, da sicherlich einige User sich für USP/MyFritz immer noch interessieren und das mit tr069 deaktivieren dann schon relevant ist bei FOS 7.50+

USP ist eine tr069 "Erweiterung". MyFritz nutzt USP, daher braucht es offenbar auch aktives tr069 als "Basisdienst", andernfalls ist
MyFritz in der aktuellen Version auch mit den Handy-Apps nicht mehr sinnvoll nutzbar. Auch Homeautomation Krempel
wird darüber gemanaged, sowie das Myfritz "dyndns" in v6 / ds-lite Umgebungen. Das hat AVM auch irgendwo beschrieben.

USP wird bei der Telekom auch für "EasySupport" noch genutzt - sofern man im Kundencenter "automatische Einrichtung/Easysupport" mit aktiviert hat. Es gibt also ein Zusammenspiel zwischen "tr069-Basisdiensten" und tr369/USP-Diensten.

Auch für mich ist "remote schnüffeln" nicht wirklich relevant und MyFritz + tr069 ist deaktiv. Im Bekanntenkreis, wo dann oftmals
nur eine Box steht, wird das aber ganz gerne mit den AVM-Apps genutzt, egal was ich von der "Sicherheit" selber davon halte (...)

Da ich es selber auch noch nicht genutzt habe, müsste ich mir den aktuellen "tr069-daemon" auch erst mal ansehen, bzw. herausfinden, welcher daemon dann genau das USP realisiert. Ich hab zumindest gesehen, das es eine tr369.cfg extra gibt, wo dann nur entsprechende "Services" dort dann configuriert sind. Die ar7.cfg hat nur die klassischen tr069-Settings drin, so wie bisher auch.

Das "reenable" das fehlenden Web-Menüpunktes nach dem USP-Abschalten habe ich testweise über die tr369.cfg wieder
zurückgesetzt + Neustart nach edit der config.

**Comment by fda77** [comment]
Also für mich ist myfritz erstmal ein dyndns Dienst von avm mit hostnamenm *.myfritz.net.
Zu diesem fällt mir zuallererst ein dass AVM wildcard Zertifikate hat und so die Möglichkeit hat jedern Benutzer via MITM zu bespitzeln.
Dann überleg ich weshlab ich diese Wanze noch nich entfernt hab und was avm noch für Überraschungen in den anderen closed-source blobs zu bieten hat

Wie bzw warum tr069 und dieses USP da mit reingehören erschliesst sich mir mit nicht so

**Reply by maverick3005** [reply]
...na weil das auf tr069 basiert und eine Erweiterung ist. (tr3 69)

Auf dem Screenshot oben kannst Du auch erkennen, welche MyFritz Dienste da noch möglich wären:
Smarthome, Remotesetup, Mesh, Diagnose, VPN, Wartung. usw. Mit den entsprechenden Android-Apps, die sich zum "AVM-Cloud-Dienst" wohl verbinden kann man dann wohl quasi die ganze Box/komplettes Mesh remote warten (oder so....)

Das wurde wohl laut diverser AVM-Infos auch favorisiert da viele User mit Ihren DS-Lite Anschlüssen und oftmals fehlendem ipv6 unterwegs nicht mehr auf Ihre heimische Box kommen. Also fungiert AVM in der Mitte jetzt als "Universal-Gateway"

Hier hast Du ein paar extra Infos noch:
https://de.wikipedia.org/wiki/TR-369

Pauschal kann man also sagen, ein typischer Telekom-Kunde hat 2 USP-Dienste aktiv:
- EasySupport der Telekom
- MyFritz von AVM

und schon rennt sein Android Krempel. Der User soll alles so remote managen können. Oder so...

Wenn ich mir die Image-configs so ansehe dann wird wohl über tr069 beim Aufbau der Verbindung übermittelt, welche weiteren
"Services" vom Anbieter noch zur Verfügung stehen und dann entsprechend tr369 dazu passend aktiviert.

Dazu haben die typischen Anbieter Ihre Zertifikate bei AVM abgegeben zum Einbau in die aktuelle Firmware.
Den Status von EasySupport on/off kann man also mit einem Telekom-Anschluss über tr069 mit abfragen.

**Comment by fda77** [comment]
Die Telekom hat sich eh Backdoors eingebaut um sich selbst ohne Zustimmung einen Remotezugriff zu erschleichen! Das ist mir den Speedports schon mehrfach während einer Störungsmeldung passiert! Wenn man dann nachfragt wird dreist gelogen man wollte das so... Daher würde ich auch bei AVM so einen Kram immer rauswerfen, man kann nicht prüfen was die eingebaut haben.

> Auf dem Screenshot oben kannst Du auch erkennen, welche MyFritz Dienste da noch möglich wären:

Bei mir geht Smarthome wunderbar ohne myfritz und tr???

**Reply by fda77** [reply]
https://github.com/Freetz-NG/freetz-ng/commit/7614341669cef847701b521a57d86256988a8aff

**Reply by maverick3005** [reply]
Suppa.

Mit dem RemoteAccess stimme ich Dir zu. Der Sinn der offenen Diskussion zu USP war eigentlich eher gedacht, die Zusammenhänge
des USP-Dienstes in Verbindung zu tr069 auch mit aufzueigen: Wird im menuconfig tr069 deaktiviert, so beeinflusst das auch tr369.

Nach einer "Migration" von 7.29 -> 7.50 wird dann eine tr369.cfg erstellt, welche dann bei der Diagnose diese AVM-Dienste am Ende der Liste noch als "enabled" anzeigt. Mit manuellem bearbeiten der config wird man das dann auch wieder los.

**Reply by fda77** [reply]
369 ist einfach ne Erweiterung von 069, daher ists okay wenn der 069 patch das 369 mit abräumt. Unterstützt das überhaupt ein Provider bereits?

Ich hab schon länger vor im Webif unter System eine extra Seite zu machen und dort alle möglichen interessanten avm Variablen anzuzeigen und mit Buttons einfach änderbar zu machen, zb
```
# ctlmgr_ctl r tr369 settings/enable
1
# ctlmgr_ctl w tr369 settings/enable 0
0
```

Dazu noch 2fa oder die updatesettings die sich ständig spontan-selbstaktivieren

**Reply by maverick3005** [reply]
Als "Provider" kenne ich die DrosselKom & als "OEMs" deren (Re)-Resellers, wobei die Tkom das für EasySupport wohl nutzt, wenn das
Endgerät bereits den neuen Standard unterstützt. Ich kann das nicht prüfen, da ich kein "Drossel-Kunde" bin.
In AT wäre das wohl auch magenta.at und auch die Telekom Austria (A1) nutzt sowas (laut Aussagen von Bekannten)

Bei Vodafone hab ich in den config Ordnern auch irgendwas bei den tr69-certs gefunden. Ich schätze mal, aufgrund der neuen .pem Files werden die "grossen" 3 da Ihre Sachen schon platziert haben, das sich zumindest die CallCenter der "Reseller" intern
alle möglichen Settings auslesen können.

Was das Thema "spontanes" reaktivieren von Updates angeht kenne ich das nur, wenn man aus der Internet-Providerliste einen Anbieter wechselt von von Provider -> "andere Provider"/manuelle Sachen, dabei werden jedes mal die AVM-Apps, NTP-Server und natürlich auch wieder das Autoupdate auf default gesetzt. Solange man beim "Provider" nichts ändert, oder gleiche Settings nochmals "saved", hatte ich noch nie an irgend einer Stelle ein "sporadisches" wieder einschalten gesehen jemals.

Hint: Man könnte die ganzen "049.tar" default Provider-Daten massiv durchpatchen das beim Umstellen auf einen anderen ISP oder erneutes Speichern bei simplen Anpassungen, wie den Zugangsdaten, "freetz conforme" Defaults und anständige Default-DNS-Server,
ordentliche NTP-Server, usw. dann jedes mal auch eingestellt werden. Wäre natürlich für etliche Länder durchzuführen für das es eine CC.tar gibt in AVM und AVME und ggfs. auch OEM-Ordner (1und1, usw.)

Oder nen "script-insert" wo man seine perönlichen Defaults einmal definiert und vor dem "save" dann diese Daten on the fly vor dem speichern in der ar7.cfg "Usergerecht" dann eingetragen werden. Dazu müsste man im Freetz dann einen extra Menüpunkt noch erstellen, wo man alle diese Sachen einmal einrichtet, inkl. auch ggfs. DHCP-Settings, ipv6-Settings mit allen Einstellungen, die sich sonst
auch mitändern/zurücksetzen würden.

Wenn du am Ende der ar7.cfg die "Update" config-Sektion auf "disable" setzt und einige weitere "yes" Optionen auf "no", so wird der Menupunkt mit den 3 Auswahloptionen beim "Autoupdate" vollständig aus dem WebIF vom FOS entfernt.
Auch im Bereich "tr069 enable = yes" kann man noch ein "no" setzen - macht auch dann Sinn wenn man den tr069 daemon sowieso entfernt hat. Dann sollte meines Wissens auch dieser Menupunkt zu tr69 verschwinden im FOS bei der "Diagnose" / Info.

**Reply by maverick3005** [reply]
Noch ein Nachtrag, was ich von alten Hybrid-Routern der Telekom & Magenta noch weiss. Speziell auch beim LTE-Teil:

Die übertragen nach wie vor ne kleine Lawine direkter "interner" Netze mit direkten "internen" Gateways um dort hin zukommen, teilweise auch eigene "intern" DNS-Server die dann in den Routingtabellen diese "Speedports" auch auf der "Eningeer"-Seite auftauchen.

Diese Routen werden auch speziell auf den DSL & LTE-Teil "gerouted" in der Kernel-Routing-Table.

Da gehts wohl um deren IPTV-Kram, bzw. auch bei VoIP-Servern werden da Net- & Hostrouten zusätzlich noch übertragen.
Ich denke das ist wohl noch für den älteren Standard mit "VLAN8" usw. relevant da immer noch alte MediaServer im Kundeneinsatz sind.
(zumindest Multicast-Ziele bekommt man über tr69 & co. ebenso noch angeliefert)

Beim Hybrid weiss ich zumindest, das wenn VoIP da auch über Mobilfunk laufen soll, das die dafür extra Routen dynamisch updaten, das das dann auch über Mobilfunk oder "Schnellstart" immer problemlos? laufen soll. Das werden dann wohl auch die IP-Bereiche sein, die aus dem "öffentlichen" Netz über andere Anbieter nicht direkt erreichbar sind. Das sind alles so seltsame Relikate und recht suspekt.

Auch an ein internes 10er "Wartungsnetz" für die CallCenter kann ich mich noch wage erinnern. Somit sind dann Speedports auch dann noch erreichbar, wenn der Kunde sich seine Zugangsdaten wohl komplett vermurxt ?

(Und laut Changelog für die FOS 7.50 & Betas ist das EasySupport jetzt auch auf die "Fremdmarke" von AVM eingebaut wurden).

Da steht wörtlich: vollständige automatische Einrichtung mit EasySupport. Damals hat es gereicht beim Usernamen noch den "anonymous"-user als Default zu belassen, der Rest wurde selber erkannt, jedoch erfolgte keine VoIP-Autoconfig. Das soll jetzt wohl auch mit provionierbar sein ? Auch hatte ich in einem Telekom-Blog gelesen, das auch das Mesh-System dann über Remote-Access für den Kunden dann ebenso managebar sein soll ? Oder Homeautomation ?

Ein Telekom-User könnte ja mal berichten, ob das mit FOS 7.50 (im Originalzustand) und aktivem EasySupport tatsächlich so ist ?
Ferner: ob man auch über den "Telekom-Login" sein HomeNet über den Provider faktisch confen kann - somit auch die CallCenter
Mitarbeiter quasi nen Vollzugriff "von hinten" rum jetzt haben....

**Reply by fda77** [reply]
Du kannst dir die ganzen ui-module via script setzen, zb in rc.custom. ctlmgr_ctl würde ich dem direkten editieren der ar7 vorziehen...

**Reply by maverick3005** [reply]
Ja, das neue Feature ist nett um das leidige tr069 = no und auch tr369.cfg enable = no, ohne langes shell murxen gleich richtig zu setzen.
Vor allem, wenn FOS das wieder reenabled (bei Anpassungen des ISPs/Wechsel), steht ja gerne sofort wieder "yes/enabled" drin.

**Comment by fda77** [comment]
es gibt zu viele 5590 issues/diskussionen, ich mache diese mal zu
